clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,793 @@
+         site_name (str): The value for 'Name' tag used for all resource
+         vpc_id (str): vpc Id to associate gateway to vpc
+    """
+    LOGGER.info("Creating Internet Gateway.")
+    internet_gateway = ec2_client.create_internet_gateway()
+    internet_gateway_id = internet_gateway["InternetGateway"]["InternetGatewayId"]
+    ec2_assign_name_tag(
+        ec2_client=ec2_client,
+        res_id=internet_gateway_id,
+        tag_name=f"{site_name}-igw",
+    )
+
+    ec2_client.attach_internet_gateway(
+        InternetGatewayId=internet_gateway_id,
+        VpcId=vpc_id,
+    )
+    return internet_gateway_id
+
+
+def allocate_elastic_ips(ec2_client, subnet_count, site_name) -> list:
+    """
+    Allocate public elastic IPs for NAT
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): subnet to alloacte elastic ip per subnet
+         site_name (str): vpc name to associate elastic ips
+
+    Returns:
+         list: list of elastic ips allocated
+    """
+
+    eip_list = []
+    for count in range(subnet_count):
+        eip = ec2_client.allocate_address(Domain="vpc")
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=eip["AllocationId"],
+            tag_name=f"{site_name}-eip{count+1}",
+        )
+        eip_list.append(eip["AllocationId"])
+    return eip_list
+
+
+def create_nat_gateway(
+    ec2_client, site_name, subnet_count, vpc_output, alloc_ids, subnet_type
+) -> list:
+    """
+    Creates NAT gateway to map private network to single public ip
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): subnet count to create nat gateway per subnet
+         site_name (str): vpc name to associate nat gateway
+
+    Returns:
+         list (nat_gateway_id_list): list containing nat gateway id
+    """
+    LOGGER.info("Creating Nat Gateway (1 per AZ) for each public subnet")
+
+    nat_gateway_id_list = []
+    for count in range(subnet_count):
+        nat_gateway = ec2_client.create_nat_gateway(
+            AllocationId=alloc_ids[count],
+            SubnetId=vpc_output["public_subnet_ids"][count],
+        )
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=nat_gateway["NatGateway"]["NatGatewayId"],
+            tag_name=f"{site_name}-nat-{subnet_type}{count+1}",
+        )
+        nat_gateway_id_list.append(nat_gateway["NatGateway"]["NatGatewayId"])
+    ec2_client.get_waiter("nat_gateway_available").wait(
+        NatGatewayIds=nat_gateway_id_list,
+    )
+    return nat_gateway_id_list
+
+
+def create_public_route_table(ec2_client, site_name, vpc_id) -> list:
+    """
+    Creates route table for public routes
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         site_name (str): vpc name
+         vpc_id (str): vpc Id of the associated vpc
+    Return:
+         str (rt_public_id): route table id
```

Comment:
`str (rt_public_id): route table id` this is not a valid format for `Returns`

Reply:
done